### PR TITLE
Refactored kernels and windows

### PR DESCRIPTION
**Pull request by cjweir**
SkipGramVectorizer and TokenCoocurrenceVectorizer now allow for string named kernel and window functions which were moved into a new seperate file.  Instead of args, the classes now simply take window_radius.

### CODE REVIEW

File: vectorizers/_vectorizers.py [hunk 1]
```diff
@@ -269,15 +271,7 @@ def preprocess_token_sequences(
...
-    """
-    result_sequences = []
-    for sequence in token_sequences:
-        result_sequence = np.array(
-            [token_dictionary[token] for token in sequence if token in token_dictionary]
-        )
-        if len(result_sequence) > 0:
-            result_sequences.append(result_sequence)
-    """
+
```
Comment (lmcinnes) [inline vectorizers/_vectorizers.py:274]: We aren't converting to integer sequences anymore? Did I miss something?

File: vectorizers/_vectorizers.py [hunk 2]
```diff
@@ -1156,6 +1126,68 @@ def fit(self, X, y=None, **fit_params):
 
 
 class SkipgramVectorizer(BaseEstimator, TransformerMixin):
+    """Given a sequence, or list of sequences of tokens, produce a
```
Comment (lmcinnes) [inline vectorizers/_vectorizers.py:1129]: Thanks for adding all the docstring!

File: vectorizers/_vectorizers.py (hunk 1 above)
Comment (cjweir) [reply vectorizers/_vectorizers.py:274]: This was already commented out and I just deleted it.  I think the pre-process_token_sequences does the conversion and so we don't need it here .

Comment (lmcinnes) [reply vectorizers/_vectorizers.py:274]: Ah, fair. Good call then.

**Pull request merged**